core/dbt/exceptions.py

```diff
@@ -2136,6 +2136,22 @@ def get_message(self) -> str:
         return msg
 
 
+class ContractError(CompilationError):
+    def __init__(self, yaml_columns, sql_columns):
+        self.yaml_columns = yaml_columns
+        self.sql_columns = sql_columns
+        super().__init__(msg=self.get_message())
+
+    def get_message(self) -> str:
+        msg = (
+            "Please ensure the name, data_type, and number of columns in your `yml` file "
+            "match the columns in your SQL file.\n"
+            f"Schema File Columns: {self.yaml_columns}\n"
+            f"SQL File Columns: {self.sql_columns}"
+        )
```

Comment:
Were you going to update this as related to https://github.com/dbt-labs/dbt-core/issues/7147?